### PR TITLE
doc: zigbee guide for adding ZCL clusters

### PR DESCRIPTION
**Pull request by greg-fer**
Added new guide for adding ZCL clusters to Zigbee app.
KRKNWK-11075.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+Use the :ref:`Zigbee Template <zigbee_template_sample>` sample as the base for adding new ZCL clusters:
+
+1. Make sure that you meet the requirements for building the sample.
+#. Build and test the sample as described on its documentation page.
+#. Copy the contents of the :file:`samples/zigbee/template` directory to a new directory meant for your custom application.
+   For example, :file:`samples/zigbee/sensor`.
+
+.. rst-class:: numbered-step
+
+Edit clusters using the ZAP tool
```
Comment (tomchy) [inline doc/nrf/ug_zigbee_adding_clusters.rst:44]: Unfortunately currently we have to do this manually...

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+Adding the functionalities for an on/off switch and a sensor requires adding new clusters.
+
+1. XYZ
+
+At this point, new clusters have been added to the Matter accessory.
+
+.. rst-class:: numbered-step
+
+Edit the main loop of the application
```
Comment (tomchy) [inline doc/nrf/ug_zigbee_adding_clusters.rst:55]: ZCL callback?

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+.. rst-class:: numbered-step
+
+Define the device role
+**********************
+
+The Zigbee template sample by default implements the Zigbee Router role.
+To change the network role, complete the following steps:
+
+1. Lorem ipsum
+#. sit etiam dolorem amat
```
Comment (greg-fer) [inline doc/nrf/ug_zigbee_adding_clusters.rst:66]: @FilipZajdel , this was requested by Krzysztof Loska at the latest sprint review.

File: doc/nrf/links.txt [hunk 4]
```diff
@@ -593,6 +606,13 @@
 .. _`Nordic Semiconductor's IoT cloud platform`: https://www.nordicsemi.com/Products/Cloud-services
 .. _`Securely Generating Credentials on the nRF9160`: https://docs.nrfcloud.com/Guides/GettingStarted/Devices#securely-generating-credentials-on-the-nrf9160
 
+.. ### Source: zigbeealliance.org & csa-iot.org
+
+.. _`Zigbee Alliance`:
+.. _`Connectivity Standards Alliance`: https://csa-iot.org
+.. _`Zigbee Specification`: https://zigbeealliance.org/developer_resources/
```
Comment (maciejpietras) [inline doc/nrf/links.txt:613]: I don't think you can find Zigbee specification under this address?

File: doc/nrf/ug_zigbee_adding_clusters.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_adding_clusters.rst]: Section removed, will be covered in https://projecttools.nordicsemi.no/jira/browse/KRKNWK-11394

File: doc/nrf/links.txt (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/links.txt:614]: There's the Zigbee BDB Specification v1.0 there. Still, it's not a new link, I've just grouped them together, so a potential change could trigger other edits outside of this PR. If you think it is wrong, I'll create a task for going through this and other Zigbee links.

**Review by maciejpietras — APPROVED** [review]

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+* Issue ZDO commands to read the list of clusters contained on a device, if you know the device's short address.
+
+Both options are described in the following sections.
+
+Looking for a specific cluster in the Zigbee network
+----------------------------------------------------
+
+You can send a match descriptor request when you want to find a specific cluster in the network.
+This request is a broadcast command that expects profile and cluster IDs as return values.
+When using the network coordinator node with the Zigbee shell enabled, you can sent the ZDO match descriptor request in the following way:
```
Comment (FilipZajdel) [inline doc/nrf/ug_zigbee_adding_clusters.rst:322]: minor: it should be "send" instead of "sent".

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+You can further expand the application with more features, such as OTA support.
+
+.. _ug_zigee_adding_clusters_ota:
+
+Adding OTA
+==========
+
+To extend the sample with OTA support, we would have to complete steps similar to :ref:`adding On/Off Switch and Temperature Sensor functionalities <ug_zigee_adding_clusters_adding_clusters>`.
+Then, we would have to implement the ZCL device callback to control the process of collecting chunks of new firmware.
+This is describe more broadly in the following sections.
```
Comment (FilipZajdel) [inline doc/nrf/ug_zigbee_adding_clusters.rst:404]: minor: "described" instead of "describe"

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+To show how you can use the ZCL event, read the following section.
+
+Reading device callback parameters from a ZBOSS buffer
+------------------------------------------------------
+
+The ZCL callback implemented in the :ref:`lib_zigbee_fota` library is a good example of how to use the ZCL event.
+
+To read the device callback parameters from a ZBOSS buffer, complete the following steps:
+
+1. In the :file:`main.c` file of the copied template sample, extend the `zb_zcl_device_callback_param_t`_ structure that contains the ZCL callback parameters in the following manner:
```
Comment (FilipZajdel) [inline doc/nrf/ug_zigbee_adding_clusters.rst:477]: The structure is some kind of a placeholder for parameters passed to callback.

I propose following sentence:
In the :file:`main.c` file of the copied template sample, use the `zb_zcl_device_callback_param_t`_ structure to get the ZCL callback parameters from the buffer in the following manner:

File: doc/nrf/ug_zigbee_adding_clusters.rst [hunk 8]
```diff
@@ -0,0 +1,506 @@
...
+
+   .. code-block:: C++
+
+      zb_zcl_device_callback_param_t *device_cb_param = ZB_BUF_GET_PARAM(bufid, zb_zcl_device_callback_param_t);
+
+   Once the parameters are obtained, the application can use them to perform some action based on a new attribute values.
+#. Make the application check the available callback IDs by exteding the `zb_zcl_device_callback_id_t`_ structure.
```
Comment (FilipZajdel) [inline doc/nrf/ug_zigbee_adding_clusters.rst:484]: What we do here is checking what kind of ZCL event has been just received. It usually differs between callbacks executions.

I propose: Make the application check the callback ID by **reading appropriate field** from the `zb_zcl_device_callback_id_t`_ structure.

**Review by FilipZajdel — CHANGES_REQUESTED** [review]

**Review by FilipZajdel — APPROVED** [review]

**Pull request merged**